dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java

```diff
@@ -652,6 +661,23 @@ private boolean processStateChangeHandler(StateEvent stateEvent) {
         return true;
     }
 
+    private boolean processBlockHandler(StateEvent stateEvent) {
+        try {
+            TaskInstance task = processService.findTaskInstanceById(stateEvent.getTaskInstanceId());
+            BlockingParameters parameters = (BlockingParameters) TaskParametersUtils.getParameters(TaskType.BLOCKING.getDesc(),
+                    task.getTaskParams());
+            if (parameters.isAlertWhenBlocking()) {
+                ProjectUser projectUser = processService.queryProjectWithUserByProcessInstanceId(processInstance.getId());
+                processAlertManager.sendProcessBlockingAlert(processInstance, projectUser);
+                logger.info("block alert send successful!");
+            }
+            taskStateChangeHandler(stateEvent);
```

Comment:
When detecting process block flag becomes true, I think it can send two events, one for task status change(blocking task runs complete), another for blocking the process. What do you think about this?

Reply:
> It's better to handle task block directly, but not call `taskStateChangeHandler`, because it's the different event type.

When detecting process block flag becomes true, I think it can send two events, one for task status change(blocking task runs complete), another for blocking the process. What do you think about this?

Reply:
It's OK, how about adding a taskState event, but not call `taskStateChangeHandler` directly?